README.md

```diff
@@ -84,17 +143,19 @@ It is costy - calculates resources for each authenticated user's request! May be
 
 # Payment providers
 
-## Website
+There is a clear distinction between "self-hosted" vs "external" subscription solutions. "Self-hosted" means that everything (plans, upgrade/downgrade rules, charging schedules etc) is set up on backend side, and payment provider is only used to store CC information and charge money when  backend asks to do so. However, some payment providers don't allow that, thus it is called "external" - everything is configured in provider dashboard, and it's a provider who does periodic charges etc. The only thing backend does - receives webhook (or periodically fetches information itself, or both altogether), and updates subscriptions (changes timespan, pauses, resumes etc).
 
-### Paddle
+"Self-hosted" provider means that backend is the command center, and it is capable of implementing custom logic of any complexity. The drawback here is that backend is responsible for periodic checks and charges, which requires monitoring and extra setup.
 
+"External" providers are limited to whatever logic is provided by third-party developers. However, it is much easier to setup and maintain it.
 
+## Paddle
 
-## Mobile apps
+Uses [paddle.com](https://paddle.com) as payment provider.
 
-Mobile apps are example of providers where most of the job is done on 3rd-party and backend only has to keep database up-to-date.
+This implementation is self-hosted. Paddle does not provide a lot of control over charges, however it has an undocumented opportunity to create a zero-cost subscription with infinite charge period. After this subscription is created, user can be charged by backend at any moment with any amount.
```

Comment:
does this mean trial subscription is not available for other payment providers?
if charging 0 is a problem, couldn't we just charge minimal amount possible for a particular provider?

Reply:
Yes and no. Google and Apple don't allow you charge yourself, the only thing that backend can do is to 1) receive payment webhooks 2) grant access to consumable based on webhook content. So if there is a trial period, it should be configured at google play / appstore by app owner.